src/vm_x86.dasc

```diff
@@ -124,16 +124,46 @@
 |.define CFRAME_SPACE,	aword*7			// Delta for esp (see <--).
 |.macro saveregs_
 |  push edi; push esi; push ebx
+|.if WIN
+|  push 0
+|  fs; push dword [0]
+|.endif
 |  sub esp, CFRAME_SPACE
 |.endmacro
 |.macro saveregs
 |  push ebp; saveregs_
 |.endmacro
 |.macro restoreregs
 |  add esp, CFRAME_SPACE
+|.if WIN
+|  fs; pop dword [0]
+|  pop edi					// Just to increment esp by 4
+|.endif
 |  pop ebx; pop esi; pop edi; pop ebp
 |.endmacro
 |
+|.if WIN
```

Comment:
Given how common user-mode callbacks from the OS are on Windows, and that the OS doesn't always give you an aligned stack, a sane compiler has to assume that functions with external linkage (or whose addresses are taken etc.) might be called with just 4-byte stack alignment (i.e. unlike Linux, the gcc/libc folk don't control all of userland, and thus cannot redefine the rules). That said, I'll accept that just maintaining stack alignment is less pain than worrying about insane compilers.